src/azul/plugins/metadata/anvil/indexer/transform.py

```diff
@@ -354,8 +355,16 @@ def _entity(self,
         field_values = ChainMap(metadata,
                                 {'document_id': entity.entity_id},
                                 additional_fields)
+
+        def field_value(field_name: str) -> AnyMutableJSON:
```

Comment:
This (_entity) is probably a hot method (called a lot). ChainMap is implemented in Python and every `[]` involves a loop. Now that you need this nested function, I think _entity can be optimized by eliminating the ChainMap (and the singleton dict for document_id instance and instead perform the lookups explicitly in the nested function. You should probably measure the optimization with a profiler.

Reply:
I profiled `TestAnvilResponse.test_entity_indices` using PyCharm's built-in profiler and found that `_entity` was called 66 times during the test, with a total execution time of 2 ms (0.0%).

Reply:
Agreed at PL to test a little bit more by using a loop to emphasize the transformation step.